packages/sdk/src/oracle/common/services/statemachine.ts

```diff
@@ -0,0 +1,266 @@
+import { Context, Memory, ContextType } from "oracle/common/types/statemachine";
```

Comment:
this file was refactored but functinality is not really changed. Only major change is that state machine handlers can be registered at run time. This is to allow different state handlers based on different oracle interfaces.